File: artplayer-java/src/main/java/org/salient/artplayer/AbsControlPanel.java
package org.salient.artplayer;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.SeekBar;

/**
 *  Created by Mai on 2018/7/10
 * *
 *  Description:视频播放控制面板的基类
 * *
 */
public abstract class AbsControlPanel extends FrameLayout implements MediaStateListener, View.OnClickListener, View.OnTouchListener, SeekBar.OnSeekBarChangeListener{

    protected VideoView mTarget;

    public AbsControlPanel(Context context) {
        super(context);
        init(context);
    }

    public AbsControlPanel(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public AbsControlPanel(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    protected abstract int getResourceId();

    protected void init(Context context) {
        View.inflate(context, getResourceId(), this);
    }

    @Override
    protected void onLayout(boolean changed, int l, int t, int r, int b) {
        super.onLayout(changed, l, t, r, b);
    }

    public void notifyStateChange() {
        switch (MediaPlayerManager.instance().getPlayerState()) {
            case ERROR:
                onStateError();
                break;
            case IDLE:
                onStateIdle();
                break;
            case PAUSED:
                onStatePaused();
                break;
            case PLAYING:
                onStatePlaying();
                break;
            case PREPARED:
                onStatePrepared();
                break;
            case PREPARING:
                onStatePreparing();
                break;
            case PLAYBACK_COMPLETED:
                onStatePlaybackCompleted();
                break;
        }
    }

    public void setTarget(VideoView target) {
        this.mTarget = target;
    }

    public VideoView getTarget() {
        return mTarget;
    }

    public void hideUI(View... views) {
        for (View view : views) {
            if (view != null) {
                view.setVisibility(INVISIBLE);
            }
        }
    }

    public void showUI(View... views) {
        for (View view : views) {
            if (view != null) {
                view.setVisibility(VISIBLE);
            }
        }
    }

    @Override
    public void onStateError() {

    }

    @Override
    public void onStateIdle() {

    }

    @Override
    public void onStatePreparing() {

    }

    @Override
    public void onStatePrepared() {

    }

    @Override
    public void onStatePlaying() {

    }

    @Override
    public void onStatePaused() {

    }

    @Override
    public void onStatePlaybackCompleted() {

    }

    @Override
    public void onSeekComplete() {

    }

    @Override
    public void onBufferingUpdate(int progress) {

    }

    @Override
    public void onInfo(int what, int extra) {

    }

    @Override
    public void onProgressUpdate(int progress, long position, long duration) {

    }

    @Override
    public void onEnterSecondScreen() {

    }

    @Override
    public void onExitSecondScreen() {

    }

    @Override
    public void onClick(View v) {

    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        return false;
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {

    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {

    }

    /**
     * 进入全屏
     * @param screenOrientation 屏幕方向
     */
    public abstract void enterFullScreen(int screenOrientation);
}

File: artplayer-kotlin/src/main/java/org/salient/artplayer/AbsControlPanel.kt
package org.salient.artplayer

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.widget.FrameLayout
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import org.salient.artplayer.MediaPlayerManager.PlayerState

/**
 * Created by Mai on 2018/7/10
 * *
 * Description:视频播放控制面板的基类
 * *
 */
abstract class AbsControlPanel : FrameLayout, MediaStateListener, View.OnClickListener, OnTouchListener, OnSeekBarChangeListener {
    var target: VideoView? = null

    constructor(context: Context?) : super(context!!) {
        init(context)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
        init(context)
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {
        init(context)
    }

    protected abstract val resourceId: Int
    protected fun init(context: Context?) {
        View.inflate(context, resourceId, this)
    }

    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
        super.onLayout(changed, l, t, r, b)
    }

    fun notifyStateChange() {
        when (MediaPlayerManager.playerState) {
            PlayerState.ERROR -> onStateError()
            PlayerState.IDLE -> onStateIdle()
            PlayerState.PAUSED -> onStatePaused()
            PlayerState.PLAYING -> onStatePlaying()
            PlayerState.PREPARED -> onStatePrepared()
            PlayerState.PREPARING -> onStatePreparing()
            PlayerState.PLAYBACK_COMPLETED -> onStatePlaybackCompleted()
        }
    }

    fun hideUI(vararg views: View?) {
        for (view in views) {
            if (view != null) {
                view.visibility = View.INVISIBLE
            }
        }
    }

    fun showUI(vararg views: View?) {
        for (view in views) {
            if (view != null) {
                view.visibility = View.VISIBLE
            }
        }
    }

    override fun onStateError() {}
    override fun onStateIdle() {}
    override fun onStatePreparing() {}
    override fun onStatePrepared() {}
    override fun onStatePlaying() {}
    override fun onStatePaused() {}
    override fun onStatePlaybackCompleted() {}
    override fun onSeekComplete() {}
    override fun onBufferingUpdate(progress: Int) {}
    override fun onInfo(what: Int, extra: Int) {}
    override fun onProgressUpdate(progress: Int, position: Long, duration: Long) {}
    override fun onEnterSecondScreen() {}
    override fun onExitSecondScreen() {}
    override fun onClick(v: View) {}
    override fun onTouch(v: View, event: MotionEvent): Boolean {
        return false
    }

    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {}
    override fun onStartTrackingTouch(seekBar: SeekBar) {}
    override fun onStopTrackingTouch(seekBar: SeekBar) {}
    /**
     * 进入全屏
     * @param screenOrientation 屏幕方向
     */
    abstract fun enterFullScreen(screenOrientation: Int)
}